src/MongoDB.Driver.Core/Core/Bindings/CoreSession.cs

```diff
@@ -482,6 +482,27 @@ private WriteConcern GetTransactionWriteConcern()
                 WriteConcern.WMajority;
         }
 
+        private WriteConcern GetCommitTransactionWriteConcern(bool isCommitARetry)
+        {
+            var writeConcern = GetTransactionWriteConcern();
+            if (isCommitARetry)
+            {
+                return writeConcern.WTimeout == null
+                    ? writeConcern.With(mode:"majority", wTimeout: TimeSpan.FromMilliseconds(10000))
+                    : writeConcern.With("majority");
+            }
+            else
+            {
+                return writeConcern;
+            }
+        }
+
+        private bool IsFirstAttemptRetry()
```

Comment:
Probably should be renamed

    IsFirstCommitAttemptARetry

To be clear about what is being attempted. (I also added an "A" because you used "A" in some but not all places).

Reply:
Thanks for catching the inconsistency! Per offline discussion, went with "A"-less version across the board.